### PR TITLE
Update readme and phpdoc for initial Optimization Detective release

### PR DESCRIPTION
**Pull request by westonruter**
This updates the `readme.txt` for Optimization Detective along with some phpdoc fixes for an initial release on the WordPress Plugin Directory. It also fixes a bad condition in a `_doing_it_wrong()` check.

The bulk of the readme changes are in a Google Doc which will be added here once reviewed.

### CODE REVIEW

File: plugins/optimization-detective/readme.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-=== Optimization Detective ===
+=== Optimization Detective (Developer Preview) ===
 
 Contributors:      wordpressdotorg
 Requires at least: 6.3
```
Comment (mukeshpanchal27) [inline plugins/optimization-detective/readme.txt]:
```suggestion
Requires at least: 6.4
```

Should be 6.4

Comment (westonruter) [reply plugins/optimization-detective/readme.txt]: Oh, I missed that in https://github.com/WordPress/performance/pull/1068.

Thanks!

File: plugins/optimization-detective/detection.php [hunk 2]
```diff
@@ -23,9 +23,9 @@ function od_get_detection_script( string $slug, OD_URL_Metrics_Group_Collection
 	/**
 	 * Filters the time window between serve time and run time in which loading detection is allowed to run.
 	 *
-	 * Allow this amount of milliseconds between when the page was first generated (and perhaps cached) and when the
+	 * This the allowance of milliseconds between when the page was first generated (and perhaps cached) and when the
```
Comment (felixarntz) [inline plugins/optimization-detective/detection.php]: Typo / missing word here.

**Review by felixarntz — APPROVED** [review]
LGTM, just one thing.

File: plugins/optimization-detective/detection.php (hunk 2 above)
Comment (westonruter) [reply plugins/optimization-detective/detection.php]: God catch

```suggestion
	 * This is the allowance of milliseconds between when the page was first generated (and perhaps cached) and when the
```

😉

**Pull request merged**